crates/compiler/builtins/roc/Crypto.roc

```diff
@@ -0,0 +1,133 @@
+module [
+    emptySha256,
+    sha256AddBytes,
+    sha256Digest,
+    hashSha256,
+    digest256ToBytes,
+    Sha256,
+    Digest256,
+]
+
+import Bool exposing [Eq]
+import List
+import Num exposing [U8, U64, U128]
+import Result
+import Str
+
+## Represents the state of a SHA-256 cryptographic hashing function, after some (or no) data has been added to the hash.
+Sha256 := { location : U64 }
+
+## Represents the digest of some data produced by the SHA-256 cryptographic hashing function as an opaque type.
+
+## `Digest256` implements the `Eq` ability.
+Digest256 := { firstHalf : U128, secondHalf : U128 } implements [Eq]
+
+## Returns an empty SHA-256 hasher.
+emptySha256 : {} -> Sha256
+
+## Adds bytes of data to be hashed by a SHA-256 hasher..
+sha256AddBytes : Sha256, List U8 -> Sha256
+
+## Returns the digest of the cryptographic hashing function represented by a SHA-256 hasher..
+sha256Digest : Sha256 -> Digest256
+
+## Applies the SHA-256 cryptographic hashing function to some bytes.
+hashSha256 : List U8 -> Digest256
+hashSha256 = \bytes -> emptySha256 {} |> sha256AddBytes bytes |> sha256Digest
+
+# Assumes little-endian. Probably shouldn't.
```

Comment:
I assume roc should be able to compile to big-endian architectures, too, Since, I can't see anyway of getting this information in roc-programmer space, I think some of this functionality will need pushing to zig